beast/tools/split_simulated_catalog.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+"""
+Split a simulated catalog and a set of AST results by number of sources
+(to improve fitting efficiency for large grids).
+"""
+import argparse
+import numpy as np
+from astropy.table import Table
+
+
+def split_main(
+    beast_settings_info,
+    catfile,
+    astfile,
+    n_per_file=1000,
+    min_n_subfile=None,
+    sort_col="F475W_RATE",
+):
+    """
+    Making the physics model grid takes a while for production runs.  This
+    creates scripts to run each subgrid as a separate job.
+
+    Parameters
+    ----------
+    beast_settings_info : string or instance
+        if string: file name with beast settings
+        if class: beast.tools.beast_settings.beast_settings instance
+
+    catfile : string
+        name of the photometry catalog file
+
+    astfile : string
+        name of the ast catalog file
+
+    n_per_file : int or None (default=6250)
+        If set, divide the split catalog into sub-catalogs with length
+        n_per_file.  Good for photometry, not useful for ASTs.
+
+    min_n_subfile : int or None (default=None)
+        If set, each bin in the photometry catalog will be split into at least
+        this many subfiles. Useful if a bin has fewer than n_per_file stars but
+        you still want flux-sorted subfiles (which means more trimming and
+        faster fitting).
+
+    sort_col : string (default="F475W_RATE")
+        If n_per_file or min_n_subfile is set, the catalog will be sorted by this
+        column before splitting into sub-catalogs.
+
+
+    """
+
+    print("Splitting catalog")
+    split_simulated_catalog(
+        catfile,
+        n_per_file=n_per_file,
+        min_n_subfile=min_n_subfile,
+        sort_col=sort_col,
+    )
+
+
+def split_simulated_catalog(
```

Comment:
I'm wondering what the point is of having 'split_main' as a function if all it does is immediately run 'split_simulated_catalog', especially since all the inputs and variable descriptions are the same, making the file 50% longer (~50 lines of code for split_main vs ~170 lines total).

Reply:
yea I set it up this way in case we need to add additional functionality. I don't mind the extra lines if it means more flexibility?